# Inferential Statistics - hypothesis Testing
Hypothesis testing is a form of statistical inference that uses data from a sample to draw conclusions about a population parameter or a population probability distribution.

In [2]:
# Import Packages
import pandas as pd
from scipy import stats
from statsmodels.stats import weightstats as stests

In [3]:
# import dataset
df = pd.read_csv("C:\\Users\\vegat\\Porto Data\\statistic\\blood_pressure.csv")

In [4]:
df.head()

patient   sex agegrp  bp_before  bp_after
0        1  Male  30-45        143       153
1        2  Male  30-45        163       170
2        3  Male  30-45        153       168
3        4  Male  30-45        153       142
4        5  Male  30-45        146       141

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 120 entries, 0 to 119
Data columns (total 5 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   patient    120 non-null    int64 
 1   sex        120 non-null    object
 2   agegrp     120 non-null    object
 3   bp_before  120 non-null    int64 
 4   bp_after   120 non-null    int64 
dtypes: int64(3), object(2)
memory usage: 4.8+ KB


In [7]:
df.shape

(120, 5)

The dataset contains 120 rows and 5 coloum data of blood pressure.

#### hypothesis Testing

Paired sampled ttest:- The paired sample testis also called dependent sample t-test. Its an uni variate test that tests for a significant ditference between 2 related variables. 
An example of this is if you where to collect the blood pressure for an individual before and after some treatment, condition, or time point.

HO :- means difference between two sample is 0

H1 :- mean diference between two sample is not 0

In [12]:
# get the value of pval of blood pressure before and blood pressure after
ttest,pval = stats.ttest_rel(df["bp_before"],df["bp_after"])
print(pval)

0.0011297914644840823


In [13]:
# Check hypotesis by comparing pval and alpha
if pval < 0.05:
    print("reject null hypotesis")
else:
    print("accept null hypotesis")

reject null hypotesis


the alpha of the code above is 5% and the hypotesis is rejected.